### PR TITLE
Delay earlier if requested

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 

## 10

---

If `DELAY_STARTUP` is set, we can delay earlier to have this option being useful for misbehaving fake hwclocks as well.

### CODE REVIEW

**Comment by dschaper** [comment]
The related Discourse issue was closed because it was cause by a user error.

**Comment by DL6ER** [comment]
It still makes sense to add the delay at the very start of FTL instead of somewhere in between (after reading the database). I see this change useful even without a related Discourse discussion.

**Comment by dschaper** [comment]
The entire reason for the delay is for networking. I think it would be quite efficient to load the database during any delay or pause.

I don't know the feasibility of that, nor unintended consequences however.

**Comment by DL6ER** [comment]
But then the delay is artificial itself. If a user specifies FTL should wait for 20 seconds, they just pick 20 because it is a nice number. I doubt they would mind if the delay was actually 20.5 seconds.

All this PR does is shifting the delay earlier so all routines replying on a properly set clock are covered as well. This seems useful to have.

File: src/main.c [hunk 1]
```diff
@@ -71,6 +71,12 @@ int main (int argc, char* argv[])
 	if(strcmp(username, "pihole") != 0)
 		logg("WARNING: Starting pihole-FTL as user %s is not recommended", username);
 
+	// Delay startup (if requested)
+	// Do this before reading the database to make this option not only
+	// useful for interfaces that aren't ready but also for fake-hwclocks
+	// which aren't ready at this point
+	delay_startup();
```
Comment (PromoFaux) [inline src/main.c:78]: Is there any value in putting this before `read_FTLconf` or even `init_shmem`, or even the first line of `main()`?

Comment (DL6ER) [reply src/main.c:78]: We need it after `read_FTLconf` because this is where we read the value how long to wait ;-)

**Comment by DL6ER** [comment]
Rebased on lastest `development`

**Comment by dschaper** [comment]
Thinking this over more, I'm going to change my mind. I still think that it would be more efficient to load the database during the delay but having the database be time dependent and the fact that the RPi boards lack a real time clock makes a delay like this logical.

And really, how much time does it take to load the database?

**Review by dschaper — APPROVED** [review]

**Pull request merged**